src/components/material-form/material-form.js

```diff
@@ -0,0 +1,364 @@
+import { Link } from 'react-router-dom';
+import TabPanel from '../tab-panel';
+import { SaveButton } from '../buttons';
+import LoadingBar from '../loading-bar';
+import useMaterialHandlers from '../../utils/use-material-handlers';
+import { useStyles } from './material-form.styles';
+import {
+  addMaterial,
+  updateMaterial,
+  showColorDialogWindow
+} from '../../redux/material/material.actions';
+import { config, routes } from '../../configs';
+import CheckboxOptions from '../checkbox-options';
+import CreateColor from '../../pages/material/create-color';
+import DialogWindowForComponent from '../dialog-window-for-component';
+
+const { languages, materialErrorMessages } = config;
+
+function MaterialForm({ material, id }) {
+  const styles = useStyles();
+  const dispatch = useDispatch();
+
+  const { loading, colors } = useSelector(({ Material }) => ({
+    loading: Material.materialLoading,
+    colors: Material.colors
+  }));
+
+  const {
+    createMaterial,
+    tabsValue,
+    handleTabsChange,
+    colorImagesToUpload,
+    setColorImagesToUpload,
+    colorImages,
+    addNewColorImages
+  } = useMaterialHandlers();
+
+  const formSchema = Yup.object().shape({
+    ukName: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    enName: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    ukDescription: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    enDescription: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    purpose: Yup.string()
+      .min(2, materialErrorMessages.MIN_LENGTH_MESSAGE)
+      .max(100, materialErrorMessages.MAX_LENGTH_MESSAGE)
+      .required(materialErrorMessages.VALIDATION_ERROR),
+
+    additionalPrice: Yup.number().required(
+      materialErrorMessages.VALIDATION_ERROR
+    )
+  });
+
+  const {
+    values,
+    handleChange,
+    handleSubmit,
+    errors,
+    touched,
+    setFieldValue
+  } = useFormik({
+    validationSchema: formSchema,
+    validateOnBlur: true,
+    initialValues: {
+      ukName: material.name[0].value || '',
+      enName: material.name[1].value || '',
+      ukDescription: material.description[0].value || '',
+      enDescription: material.description[1].value || '',
+      purpose: material.purpose || '',
+      available: material.available || false,
+      additionalPrice: 0
+    },
+    onSubmit: (data) => {
+      const newMaterial = createMaterial(data);
+
+      if (material.purpose) {
```

Comment:
What if we have a purpose but the material is not created yet? Is it possible? I think it is better to use the approach that we use for contacts where we have one dumb form `Contact-Form` and two smart components - `Contact-Add` and `Contact-Edit` to pass specific data to the form component depending on the type of operation

Reply:
it is already done so in material - one material-form is used both in material-add and material-edit.

Reply:
improved checking